src/marshmallow/fields.py

```diff
@@ -1901,30 +1873,58 @@ class EnumValue(Field):
         "unknown": "Must be one of: {choices}.",
     }
 
-    def __init__(self, cls_or_instance: Field | type, enum: type[Enum], **kwargs):
+    def __init__(
+        self,
+        enum: type[EnumType],
+        *,
+        by_value: bool | Field | type = False,
+        **kwargs,
+    ):
         super().__init__(**kwargs)
-        try:
-            self.field = resolve_field_instance(cls_or_instance)
-        except FieldInstanceResolutionError as error:
-            raise ValueError(
-                "The enum field must be a subclass or instance of "
-                "marshmallow.base.FieldABC."
-            ) from error
         self.enum = enum
-        self.choices = ", ".join(
-            [str(self.field._serialize(m.value, None, None)) for m in enum]
-        )
+        self.by_value = by_value
+
+        # Serialization by name
+        if by_value is False:
+            self.field: Field = String()
+            self.choices = ", ".join(
+                [str(self.field._serialize(m, None, None)) for m in enum.__members__]
+            )
+        # Serialization by value
+        else:
+            if by_value is True:
+                self.field = Field()
+            else:
+                try:
+                    self.field = resolve_field_instance(by_value)
+                except FieldInstanceResolutionError as error:
+                    raise ValueError(
+                        '"field" must be a subclass or instance of '
+                        "marshmallow.base.FieldABC."
+                    ) from error
+            self.choices = ", ".join(
+                [str(self.field._serialize(m.value, None, None)) for m in enum]
+            )
 
     def _serialize(self, value, attr, obj, **kwargs):
         if value is None:
             return None
-        return self.field._serialize(value.value, attr, obj, **kwargs)
+        if self.by_value:
+            val = value.value
+        else:
+            val = value.name
+        return self.field._serialize(val, attr, obj, **kwargs)
 
     def _deserialize(self, value, attr, data, **kwargs):
-        value = self.field._deserialize(value, attr, data, **kwargs)
+        val = self.field._deserialize(value, attr, data, **kwargs)
+        if self.by_value:
+            try:
+                return self.enum(val)
+            except ValueError as exc:
+                raise self.make_error("unknown", choices=self.choices) from exc
         try:
-            return self.enum(value)
-        except ValueError as exc:
+            return getattr(self.enum, val)
+        except AttributeError as exc:
             raise self.make_error("unknown", choices=self.choices) from exc
```

Comment:
nit: naming consistency

```suggestion
        except AttributeError as error:
            raise self.make_error("unknown", choices=self.choices) from error
```